From 335eb8f765591d032529eadd3bc63775102d1714 Mon Sep 17 00:00:00 2001
From: Mathew McBride <matt@traverse.com.au>
Date: Mon, 17 Aug 2026 01:41:32 +0000
Subject: [PATCH] mac80211: import "130-disable_auto_vif.patch" from owrt

mac80211: disable automatically created sta interfaces

They are not useful for anything on the system and are annoying when a
PHY is disabled in the config

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---
 net/mac80211/main.c | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/net/mac80211/main.c b/net/mac80211/main.c
index 655a8aec7738..75f91fbd6f89 100644
--- a/net/mac80211/main.c
+++ b/net/mac80211/main.c
@@ -1597,24 +1597,6 @@ int ieee80211_register_hw(struct ieee80211_hw *hw)
 
 	ieee80211_check_wbrf_support(local);
 
-	rtnl_lock();
-	wiphy_lock(hw->wiphy);
-
-	/* add one default STA interface if supported */
-	if (local->hw.wiphy->interface_modes & BIT(NL80211_IFTYPE_STATION) &&
-	    !ieee80211_hw_check(hw, NO_AUTO_VIF)) {
-		struct vif_params params = {0};
-
-		result = ieee80211_if_add(local, "wlan%d", NET_NAME_ENUM, NULL,
-					  NL80211_IFTYPE_STATION, &params);
-		if (result)
-			wiphy_warn(local->hw.wiphy,
-				   "Failed to add default virtual iface\n");
-	}
-
-	wiphy_unlock(hw->wiphy);
-	rtnl_unlock();
-
 #ifdef CONFIG_INET
 	local->ifa_notifier.notifier_call = ieee80211_ifa_changed;
 	result = register_inetaddr_notifier(&local->ifa_notifier);
